tools/sklearn/main_macros.xml

```diff
@@ -1105,6 +1173,47 @@ def get_X_y(params, file1, file2):
     <param argument="pre_dispatch" type="@TYPE@" value="@DEFAULT_VALUE@" optional="true" label="pre_dispatch" help="@HELP@"/>
   </xml>
 
+  <xml name="search_cv_estimator">
+    <conditional name="estimator_config">
+        <param name="estimator_or_pipeline" type="select" label="Choose to configure an estimator or load from a built pipeline:">
+            <option value="configure_an_estimator">Estimator - No pipeline</option>
+            <option value="load_pipeline">Pipeline - A list of estimators which could include pre-processor and feature selector</option>
+        </param>
+        <when value="configure_an_estimator">
+            <expand macro="estimator_input_no_fit" />
+            <section name="search_params_builder" title="Search parameters Builder" expanded="true">
+                <repeat name="estimator_param_set" min="1" max="10" title="Estimator parameter for search:">
+                    <param name="estimator_param" type="text" value="" optional="true" label="Estimator parameter" help="Optional. For example: C: [1, 10, 100, 1000]"/>
+                </repeat>
+            </section>
+        </when>
+        <when value="load_pipeline">
+            <param name="infile_pipeline" type="data" format="zip" label="Choose the dataset containing pipeline object:"/>
+            <section name="search_params_builder" title="Search parameters Builder" expanded="true">
+                <repeat name="prep_param_set" min="1" max="5" title="Pre_processing parameter for search:">
+                    <param name="prep_param" type="text" value="" optional="true" label="Pre_processing parameter" help="Optional. For example: with_centering: [True, False]"/>
+                </repeat>
+                <repeat name="fs_param_set" min="1" max="10" title="Feature selection parameter for search:">
+                    <param name="fs_param" type="text" value="" optional="true" label="Feature selection parameter" help="Optional. For example: k: [3, 5, 7, 9]"/>
+                </repeat>
+                <repeat name="estimator_param_set" min="1" max="10" title="Estimator parameter for search:">
+                    <param name="estimator_param" type="text" value="" optional="true" label="Estimator parameter" help="Optional. For example: C: [1, 10, 100, 1000]"/>
+                </repeat>
+            </section>
+        </when>
+    </conditional>
+  </xml>
+
+  <xml name="search_cv_options">
+      <expand macro="scoring"/>
+      <expand macro="model_validation_common_options"/>
+      <expand macro="pre_dispatch" value="2*n_jobs" help="Controls the number of jobs that get dispatched during parallel execution"/>
```

Comment:
pre_dispatch --> I'm not sure this should be exposed to the user. The user has no clue on which cluster is is scheduled how much nodes you have available and so on ...

Reply:
We should hide this parameter tpye="hidden" und just hard code it ...

Reply:
Interesting topic

Reply:
You could pull it from the environment. Then administrators could supply a value and change it as they like.